src/azul/chalice.py

```
@@ -504,34 +497,105 @@ def swagger_resource(self, file) -> Response:
+
+        https://docs.aws.amazon.com/lambda/latest/dg/invocation-retries.html
+        """
+        value: int
+        handler_name: str | None = attr.ib(default=None)
+
+        def __call__(self, f):
+            assert isinstance(f, chalice.app.EventSourceHandler), f
+            f.retries = self.value
+            return f
 
     @property
-    def metric_thresholds(self) -> list[MetricThreshold]:
-        default_threshold = 0
-        thresholds = []
+    def metric_alarms(self) -> list[metric_alarm]:
+        alarms = []
         lambda_name, _ = config.unqualified_resource_name(self.app_name)
+
         for metric in LambdaMetric:
-            # The api_handler lambda functions (indexer & service) aren't included
-            # in the app_module's handler_map, so we account for those first.
-            thresholds.append(MetricThreshold(lambda_name=lambda_name,
-                                              metric=metric,
-                                              value=default_threshold))
-            for handler_name, handler in self.handler_map.items():
-                if isinstance(handler, chalice.app.EventSourceHandler):
-                    threshold = getattr(handler, f'{metric.name}_threshold', default_threshold)
-                    thresholds.append(MetricThreshold(lambda_name=lambda_name,
-                                                      handler_name=handler_name,
-                                                      metric=metric,
-                                                      value=threshold))
-        return thresholds
+            # The api_handler lambda functions (indexer & service) aren't
+            # included in the app_module's handler_map, so we account for those
+            # first.
+            alarms.append(self.metric_alarm(metric=metric).bind(lambda_name))
+        for handler_name, handler in self.handler_map.items():
+            if isinstance(handler, chalice.app.EventSourceHandler):
```

Seems like an odd smell to silently ignore handlers with other types. If that really is the correct behavior, consider adding an `else: pass` to this `if`. Otherwise, chance to `require` or `assert`.